D403 not triggering when it should

I am seeing D4-3 not hitting a number of cases where flake8 is.
To replicate `ruff test.py --select D403 -n --isolated`

Code:
```python
"""Contains tests for the configuration library."""
import unittest


class BasicConfigurationTest(unittest.TestCase):
    """Doc string."""

    @classmethod
    def setUpClass(cls):
        """runs just once when the class instance is setup."""
        pass
```

I expect the docstring for setUpClass to catch the lower case runs, which is a D403 error

Flake8 catches the error:
``` 
poetry run flake8 test.py
test.py:10:1: D403 First word of the first line should be properly capitalized
```

I am using ruff version
poetry run ruff --version
ruff 0.0.247

pyproject.toml setup
```toml
[tool.ruff.pycodestyle]
max-doc-length = 120

[tool.ruff]
line-length = 120
exclude = [".venv"]
ignore = [
    "D105",
    "D107",
    "D401",
    "E501",
    "W505",
]
select = [
    "B",
    # "C",
    "D",
    "D415",
    "E",
    "F",
    "N",
    "W",
]

[tool.ruff.pydocstyle]
convention = "google"

[tool.ruff.per-file-ignores]
"dejavu/__init__.py" = ["F401"]
```

<!--
Thank you for taking the time to report an issue! We're glad to have you involved with Ruff.

If you're filing a bug report, please consider including the following information:

* A minimal code snippet that reproduces the bug.
* The command you invoked (e.g., `ruff /path/to/file.py --fix`), ideally including the `--isolated` flag.
* The current Ruff settings (any relevant sections from your `pyproject.toml`).
* The current Ruff version (`ruff --version`).
-->

--- a/crates/ruff/src/rules/pydocstyle/rules/capitalized.rs
+++ b/crates/ruff/src/rules/pydocstyle/rules/capitalized.rs
@@ -18,7 +18,10 @@ impl Violation for FirstLineCapitalized {
 
 /// D403
 pub fn capitalized(checker: &mut Checker, docstring: &Docstring) {
-    if !matches!(docstring.kind, DefinitionKind::Function(_)) {
+    if !matches!(
+        docstring.kind,
+        DefinitionKind::Function(_) | DefinitionKind::NestedFunction(_) | DefinitionKind::Method(_)
+    ) {
         return;
     }
 

